src/builds/travis.rs

```diff
@@ -0,0 +1,97 @@
+use std::io::Read;
+
+use chrono::{DateTime, UTC};
+use diesel;
+use diesel::prelude::*;
+use hyper::Client;
+use hyper::header::UserAgent;
+use hyper::net::HttpsConnector;
+use hyper_native_tls::NativeTlsClient;
+use serde::de::DeserializeOwned;
+use serde_json;
+
+use DB_POOL;
+use domain::builds::Build;
+use error::DashResult;
+
+header! { (Accept, "Accept") => [String] }
+
+const ROOT_URL: &'static str = "https://api.travis-ci.org";
+const UA: &'static str = "rusty-dash/0.0.0";
+
+pub fn get_and_insert_build(build: &str) -> DashResult<()> {
+    let url = format!("{}/repos/rust-lang/rust/builds/{}", ROOT_URL, build);
+    let response: ResponseFromJson = get(&url)?;
+    let conn = &*DB_POOL.get()?;
+
+    for job in response.jobs.iter() {
+        if job.finished_at.is_none() || job.state == "canceled" {
+            continue
+        }
+
+        let duration = job.finished_at.unwrap()
+            .signed_duration_since(job.started_at.unwrap());
+        let b = Build {
+            number: response.build.id,
+            builder_name: "travis".to_string(),
+            builder_os: job.config.os.clone(),
+            builder_env: job.config.env.clone(),
+            successful: job.state == "passed",
+            message: String::new(),
+            duration_secs: Some(duration.num_seconds() as i32),
+            start_time: job.started_at.map(|dt| dt.naive_utc()),
+            end_time: job.finished_at.map(|dt| dt.naive_utc()),
```

Comment:
This makes me think something similar to the AppVeyor code -- if we're going to unconditionally insert a `Some` for `duration_secs` then we should do so for start_time and end_time too, using something typesafe to prevent future mistakes.